### ISSUE TITLE
Unwinnable Seed in v2.9 ?

### ISSUE THREAD

**Issue by bailli** [thezerothcat/LaMulanaRandomizer#70]
I think I generated an unwinnable seed - or there is a path I can take that I don't know about.

Basically the right half of the Chamber of Birth is unreachable? It has 3 entrances:

> Inferno Cavern (Darkness Sanctuary) <= Amphisbaena => Chamber of Birth
> 
> Gate of Illusion (Children's Room) <==========> Chamber of Birth (Vishnu's Room)
> Chamber of Birth (Saraswati's Room) ===========> Inferno Cavern (Anterior Chamber)

1. The backside door connects to Inferno Cavern - which itself is only reachable through the door.
2. The ladder connects to Inferno Cavern as well - and is a one way transition into the lava
3. The last entrance is the door which connects the two halves after defeating Skanda. In this seed it connects to the Gate of Illusion part which is (initially) only reachable through that door until you open the "hand" shortcut... (and additionally is the place where the Mudmen could be activated to defeat Skanda in the first place.

Am I missing something?

Here are some logs for reference:

<details><summary>randomizer-config.txt</summary>

```
version=2.9.0
seed=1603102306
shopRandomization=EVERYTHING
automaticHardmode=false
requireSoftwareComboForKeyFairy=true
requireFullAccess=false
requireIceCapeForLava=false
requireFlaresForExtinction=true
randomizeXmailer=true
randomizeForbiddenTreasure=true
htFullRandom=false
randomizeDracuetShop=false
randomizeCoinChests=true
randomizeTrapItems=true
randomizeEscapeChest=false
allowWhipStart=true
allowMainWeaponStart=true
allowSubweaponStart=true
subweaponOnlyLogic=true
removeMainWeapons=false
randomizeCursedChests=true
randomizeTransitionGates=true
randomizeOneWayTransitions=true
randomizeBacksideDoors=true
replaceMapsWithWeights=false
coinChestGraphics=true
automaticGrailPoints=true
automaticTranslations=true
ushumgalluAssist=true
laMulanaBaseDir=/home/user/.games/LaMulana/
laMulanaSaveDir=/home/user/.local/share/nigoro/la-mulana/save/
language=en
bossDifficulty=MEDIUM
minRandomRemovedItems=0
maxRandomRemovedItems=0
randomization.Anchor=RANDOM
randomization.Angel Shield=RANDOM
randomization.Ankh Jewel (Gate of Guidance)=RANDOM
randomization.Ankh Jewel (Mausoleum of the Giants)=RANDOM
randomization.Ankh Jewel (Temple of the Sun)=RANDOM
randomization.Ankh Jewel (Spring in the Sky)=RANDOM
randomization.Ankh Jewel (Tower of Ruin)=RANDOM
randomization.Ankh Jewel (Chamber of Birth)=RANDOM
randomization.Ankh Jewel (Twin Labyrinths)=RANDOM
randomization.Ankh Jewel (Dimensional Corridor)=RANDOM
randomization.Axe=RANDOM
randomization.beolamu.exe=RANDOM
randomization.Birth Seal=RANDOM
randomization.Bomb=RANDOM
randomization.Book of the Dead=RANDOM
randomization.bounce.exe=RANDOM
randomization.Bracelet=RANDOM
randomization.Bronze Mirror=RANDOM
randomization.Buckler=RANDOM
randomization.bunemon.exe=RANDOM
randomization.bunplus.com=RANDOM
randomization.Caltrops=RANDOM
randomization.capstar.exe=RANDOM
randomization.Chain Whip=RANDOM
randomization.Chakram=RANDOM
randomization.Cog of the Soul=RANDOM
randomization.Crucifix=RANDOM
randomization.Crystal Skull=RANDOM
randomization.Death Seal=RANDOM
randomization.deathv.exe=RANDOM
randomization.Diary=RANDOM
randomization.Dimensional Key=RANDOM
randomization.Djed Pillar=RANDOM
randomization.Dragon Bone=RANDOM
randomization.Earth Spear=RANDOM
randomization.emusic.exe=RANDOM
randomization.Eye of Truth=RANDOM
randomization.Fairy Clothes=RANDOM
randomization.Fake Silver Shield=RANDOM
randomization.Feather=RANDOM
randomization.Flail Whip=RANDOM
randomization.Flare Gun=RANDOM
randomization.Fruit of Eden=RANDOM
randomization.Gauntlet=RANDOM
randomization.Glove=RANDOM
randomization.Grapple Claw=RANDOM
randomization.guild.exe=RANDOM
randomization.Hand Scanner=RANDOM
randomization.Heatproof Case=RANDOM
randomization.Helmet=RANDOM
randomization.Hermes' Boots=RANDOM
randomization.Holy Grail=STARTING
randomization.Ice Cape=RANDOM
randomization.Isis' Pendant=RANDOM
randomization.Katana=RANDOM
randomization.Key of Eternity=RANDOM
randomization.Key Sword=RANDOM
randomization.Knife=RANDOM
randomization.Lamp of Time=RANDOM
randomization.lamulana.exe=RANDOM
randomization.Life Seal=RANDOM
randomization.mantra.exe=RANDOM
randomization.Magatama Jewel=RANDOM
randomization.Map (Chamber of Birth)=RANDOM
randomization.Map (Chamber of Extinction)=RANDOM
randomization.Map (Dimensional Corridor)=RANDOM
randomization.Map (Endless Corridor)=RANDOM
randomization.Map (Gate of Guidance)=RANDOM
randomization.Map (Gate of Illusion)=RANDOM
randomization.Map (Graveyard of the Giants)=RANDOM
randomization.Map (Inferno Cavern)=RANDOM
randomization.Map (Mausoleum of the Giants)=RANDOM
randomization.Map (Shrine of the Mother)=RANDOM
randomization.Map (Spring in the Sky)=RANDOM
randomization.Map (Surface)=RANDOM
randomization.Map (Temple of Moonlight)=RANDOM
randomization.Map (Temple of the Sun)=RANDOM
randomization.Map (Tower of Ruin)=RANDOM
randomization.Map (Tower of the Goddess)=RANDOM
randomization.Map (Twin Labyrinths)=RANDOM
randomization.Maternity Statue=RANDOM
randomization.mekuri.exe=RANDOM
randomization.Mini Doll=RANDOM
randomization.miracle.exe=RANDOM
randomization.mirai.exe=RANDOM
randomization.Mobile Super X2=RANDOM
randomization.move.exe=RANDOM
randomization.Mulana Talisman=RANDOM
randomization.Origin Seal=RANDOM
randomization.Pepper=RANDOM
randomization.Perfume=RANDOM
randomization.Philosopher's Ocarina=RANDOM
randomization.Pistol=RANDOM
randomization.Plane Model=RANDOM
randomization.Pochette Key=RANDOM
randomization.randc.exe=RANDOM
randomization.reader.exe=RANDOM
randomization.Ring=RANDOM
randomization.Rolling Shuriken=RANDOM
randomization.Sacred Orb (Surface)=RANDOM
randomization.Sacred Orb (Gate of Guidance)=RANDOM
randomization.Sacred Orb (Mausoleum of the Giants)=RANDOM
randomization.Sacred Orb (Temple of the Sun)=RANDOM
randomization.Sacred Orb (Spring in the Sky)=RANDOM
randomization.Sacred Orb (Chamber of Extinction)=RANDOM
randomization.Sacred Orb (Twin Labyrinths)=RANDOM
randomization.Sacred Orb (Shrine of the Mother)=RANDOM
randomization.Sacred Orb (Tower of Ruin)=RANDOM
randomization.Sacred Orb (Dimensional Corridor)=RANDOM
randomization.Scalesphere=RANDOM
randomization.Scriptures=RANDOM
randomization.Serpent Staff=RANDOM
randomization.Shell Horn=RANDOM
randomization.Shuriken=RANDOM
randomization.Silver Shield=RANDOM
randomization.Spaulder=RANDOM
randomization.Talisman=RANDOM
randomization.torude.exe=RANDOM
randomization.Treasures=RANDOM
randomization.Twin Statue=RANDOM
randomization.Vessel=RANDOM
randomization.Waterproof Case=RANDOM
randomization.Woman Statue=RANDOM
randomization.yagomap.exe=RANDOM
randomization.yagostr.exe=RANDOM
glitches.Lamp Glitch=false
glitches.Cat Pause=false
glitches.Raindrop=false
glitches.Ice Raindrop=false
glitches.Pot Clip=false
glitches.Object Zip=false
dboost.Enemy=false
dboost.Item=false
dboost.Environment=false
```

</details>

<details><summary>doors.txt</summary>

```
Inferno Cavern (Darkness Sanctuary) <= Amphisbaena => Chamber of Birth
Chamber of Extinction (Approach to the Sanctuary) <= Sakit => Gate of Illusion
Temple of the Sun <= Ellmac => Tower of Ruin (Nüwa's Pyramid)
Tower of the Goddess <= Bahamut => Temple of Moonlight
Mausoleum of the Giants <= Viy => Tower of Ruin (Tower of Babel)
Inferno Cavern (Echidna's Chamber) <= Palenque => Surface
Gate of Guidance <= Baphomet => Graveyard of the Giants
```

</details>

<details><summary>gates.txt</summary>

```
Surface (Entrance to La-Mulana Ruins) <==========> Temple of the Sun (Isis' Anterior Chamber)
Surface (Sound Canyon) <==========> Shrine of the Mother (Amphisbaena's Room)
Surface (Underground Passage) <==========> Mausoleum of the Giants (Last Shrine)
Gate of Guidance (Pillars Gate) <==========> Chamber of Birth (Skanda's Room)
Gate of Guidance (Room of Courage) <==========> Gate of Time (Mausoleum of the Giants to Tower of the Goddess)
Gate of Guidance (Monument of Time) <==========> Temple of Moonlight (Stairway of Eden)
Gate of Guidance (Cliff of Radiance) <==========> Endless Corridor (Second Endless Corridor)
Mausoleum of the Giants (Moon Palace of the Giants) <==========> Graveyard of the Giants (Second Tomb)
Mausoleum of the Giants (Last Shrine) <==========> Surface (Underground Passage)
Mausoleum of the Giants (Star Palace of the Giants) <==========> Temple of the Sun (Watchtower)
Temple of the Sun (Isis' Anterior Chamber) <==========> Surface (Entrance to La-Mulana Ruins)
Temple of the Sun (Altar of Mirrors) (Upper) <==========> Chamber of Extinction (Mural of Invocation)
Temple of the Sun (Altar of Mirrors) (Lower) <==========> Chamber of Extinction (Mural of Invocation)
Temple of the Sun (Watchtower) <==========> Mausoleum of the Giants (Star Palace of the Giants)
Spring in the Sky (Mural of Oannes) <==========> Chamber of Extinction (Lake of Life)
Inferno Cavern (Statue of the Giant) <==========> Temple of Moonlight (Sealed Approach)
Inferno Cavern (Twin's Footstool) <==========> Gate of Illusion (Fool's Confusion Corridor)
Inferno Cavern (Echidna's Chamber) <==========> Shrine of the Mother (Bahamut's Room)
Chamber of Extinction (Mural of Invocation) <==========> Temple of the Sun (Altar of Mirrors) (Upper)
Chamber of Extinction (Mural of Invocation) <==========> Temple of the Sun (Altar of Mirrors) (Lower)
Chamber of Extinction (Lake of Life) <==========> Spring in the Sky (Mural of Oannes)
Chamber of Extinction (Gate of Ox-Head and Horse-Face)  <==========> Shrine of the Mother (Treasury) (Death Seal)
Chamber of Extinction (Shiva's Crucifix)  <==========> Twin Labyrinths (Poseidon's Room)
Twin Labyrinths (Twin's Surrounding Wall) <==========> Dimensional Corridor
Twin Labyrinths (Hero's Approach) <==========> Chamber of Birth (Deva's Room)
Twin Labyrinths (Sanctuary) <==========> Shrine of the Mother (Treasury) (Map)
Twin Labyrinths (Idigna's Room) ===========> Tower of the Goddess (Gate of Time exit)
Twin Labyrinths (Poseidon's Room) <==========> Chamber of Extinction (Shiva's Crucifix) 
Endless Corridor (First Endless Corridor) <==========> Tower of the Goddess (Path to Life)
Endless Corridor (Second Endless Corridor) <==========> Gate of Guidance (Cliff of Radiance)
Endless Corridor (Fifth Endless Corridor) <==========> Tower of the Goddess (Verdandi's Tower)
Shrine of the Mother (Amphisbaena's Room) <==========> Surface (Sound Canyon)
Shrine of the Mother (Bahamut's Room) <==========> Inferno Cavern (Echidna's Chamber)
Shrine of the Mother (Treasury) (Death Seal) <==========> Chamber of Extinction (Gate of Ox-Head and Horse-Face) 
Shrine of the Mother (Treasury) (Map) <==========> Twin Labyrinths (Sanctuary)
Gate of Illusion (Children's Room) <==========> Chamber of Birth (Vishnu's Room)
Gate of Illusion (Sacrificial Abyss) <==(Pipe)==> Gate of Time (Gate of Guidance to Surface)
Gate of Illusion (Fool's Confusion Corridor) <==========> Inferno Cavern (Twin's Footstool)
Gate of Illusion (Entrance of Illusion) <==========> Graveyard of the Giants (First Tomb)
Graveyard of the Giants (Third Tomb) <==========> Tower of Ruin (Burial Chamber of Life)
Graveyard of the Giants (Second Tomb) <==========> Mausoleum of the Giants (Moon Palace of the Giants)
Graveyard of the Giants (First Tomb) <==========> Gate of Illusion (Entrance of Illusion)
Graveyard of the Giants (Altar of Knowledge) <==========> Tower of the Goddess (Path to Power)
Graveyard of the Giants (Anterior Corridor) <==========> Gate of Time (Mausoleum of the Giants to Gate of Guidance)
Temple of Moonlight (Sealed Approach) <==========> Inferno Cavern (Statue of the Giant)
Temple of Moonlight (Stairway of Eden) <==========> Gate of Guidance (Monument of Time)
Temple of Moonlight (Neptune's Feet) <==========> Gate of Time (Gate of Guidance to Mausoleum of the Giants)
Tower of the Goddess (Path to Life) <==========> Endless Corridor (First Endless Corridor)
Tower of the Goddess (Urd's Tower) <==========> Tower of Ruin (Fountain of Heat)
Tower of the Goddess (Verdandi's Tower) <==========> Endless Corridor (Fifth Endless Corridor)
Tower of the Goddess (Path to Power) <==========> Graveyard of the Giants (Altar of Knowledge)
Tower of Ruin (Approach of the Philosophers) <==========> Gate of Time (Surface to Gate of Guidance)
Tower of Ruin (Burial Chamber of Life) <==========> Graveyard of the Giants (Third Tomb)
Tower of Ruin (Fountain of Heat) <==========> Tower of the Goddess (Urd's Tower)
Chamber of Birth (Vishnu's Room) <==========> Gate of Illusion (Children's Room)
Chamber of Birth (Skanda's Room) <==========> Gate of Guidance (Pillars Gate)
Chamber of Birth (Saraswati's Room) ===========> Inferno Cavern (Anterior Chamber)
Chamber of Birth (Deva's Room) <==========> Twin Labyrinths (Hero's Approach)
Dimensional Corridor <==========> Twin Labyrinths (Twin's Surrounding Wall)
Gate of Time (Mausoleum of the Giants to Gate of Guidance) <==========> Graveyard of the Giants (Anterior Corridor)
Gate of Time (Mausoleum of the Giants to Tower of the Goddess) <==========> Gate of Guidance (Room of Courage)
Gate of Time (Gate of Guidance to Mausoleum of the Giants) <==========> Temple of Moonlight (Neptune's Feet)
Gate of Time (Gate of Guidance to Surface) <==(Pipe)==> Gate of Illusion (Sacrificial Abyss)
Gate of Time (Surface to Gate of Guidance) <==========> Tower of Ruin (Approach of the Philosophers)
```

</details>

**Comment by thezerothcat** [comment]
Looking into it now, thanks!

**Comment by thezerothcat** [comment]
Access to that section of Chamber of Birth was considered to be dependent on accessing Skanda's side of the door instead of the correct one. The logic will be fixed in 2.10.0 later today. Thanks for letting me know!
